### PR TITLE
Fixing Helm Ingress Capabilities Check

### PR DESCRIPTION
**Pull request by bdun1013**
Fixing what was attempted to be fixed in https://github.com/DefectDojo/django-DefectDojo/pull/6759

The original attempted fix checked for the API version plus the resource, when it should have only checked for the API version.

### CODE REVIEW

**Review by cneill — APPROVED** [review]

**Comment by cneill** [comment]
Could you please bump the chart version for this change? Thanks

https://github.com/DefectDojo/django-DefectDojo/blob/b3d3c426deefc74e05795f2d1f30599417d7a96c/helm/defectdojo/Chart.yaml#L5

**Comment by bdun1013** [comment]
> Could you please bump the chart version for this change? Thanks
> 
> https://github.com/DefectDojo/django-DefectDojo/blob/b3d3c426deefc74e05795f2d1f30599417d7a96c/helm/defectdojo/Chart.yaml#L5

Bumped to `1.6.89`

File: helm/defectdojo/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "2.26.4"
 description: A Helm chart for Kubernetes to install DefectDojo
 name: defectdojo
-version: 1.6.88
+version: 1.6.89
 icon: https://www.defectdojo.org/img/favicon.ico
 maintainers:
```
Comment (Maffooch) [inline helm/defectdojo/Chart.yaml:7]: The versioning here is managed by the release pipeline. Please do not update it here

**Review by Maffooch — CHANGES_REQUESTED** [review]
Please base your PR towards `dev` instead of `master`

**Comment by dsever** [comment]
> Fixing what was attempted to be fixed in #6759
> 
> The original attempted fix checked for the API version plus the resource, when it should have only checked for the API version.

Can you elaborate?
Idea was was to check if resource is in beta specification or in stable to support older clusters. If you remove exact resource type then whole thing makes no sense. 

Do you some issues?

**Comment by bdun1013** [comment]
@dsever Sorry for the late reply. I think the Helm documentation is misleading. The `Capabilities.APIVersions.Has` function appears to only work properly with API versions (networking.k8s.io/v1) and not Resources (networking.k8s.io/v1/Ingress).

```
$ kubectl version       
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.2
```

```
$ helm version
version.BuildInfo{Version:"v3.13.0", GitCommit:"825e86f6a7a38cef1112bfa606e4127a706749b1", GitTreeState:"clean", GoVersion:"go1.21.1"}
```

I have the `networking.k8s.io/v1` API version available, but not the `networking.k8s.io/v1beta1` API version:
```
$ kubectl api-versions | grep networking                                                                                                
networking.k8s.io/v1
```

With the latest version of the chart, I get an Ingress resource with the `networking.k8s.io/v1beta1` API version:
```
$ helm template defectdojo/defectdojo --version 1.6.89 -s templates/django-ingress.yaml
---
# Source: defectdojo/templates/django-ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: release-name-defectdojo
  labels:
    defectdojo.org/component: django
    app.kubernetes.io/name: defectdojo
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    helm.sh/chart: defectdojo-1.6.89
spec:
  tls:
  - hosts:
    - defectdojo.default.minikube.local
    secretName: defectdojo-tls
  rules:
  - host: defectdojo.default.minikube.local
    http:
      paths:
        - path: /
          backend:
            serviceName: release-name-defectdojo-django
            servicePort: http
```

With my change, I get the correct Ingress resource:
```
# fixing_helm_ingress_capabilities_check branch
$ helm template helm/defectdojo -s templates/django-ingress.yaml
---
# Source: defectdojo/templates/django-ingress.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: release-name-defectdojo
  labels:
    defectdojo.org/component: django
    app.kubernetes.io/name: defectdojo
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    helm.sh/chart: defectdojo-1.6.90-dev
spec:
  tls:
  - hosts:
    - defectdojo.default.minikube.local
    secretName: defectdojo-tls
  rules:
  - host: defectdojo.default.minikube.local
    http:
      paths:
        - pathType: Prefix
          path: "/"
          backend:
            service:
              name: release-name-defectdojo-django
              port:
                name: http
```

**Comment by bdun1013** [comment]
> Please base your PR towards `dev` instead of `master`

@Maffooch I based the PR towards `dev`

**Comment by dsever** [comment]
> $ helm template defectdojo/defectdojo --version 1.6.89 -s templates/django-ingress.yaml
> ---
> # Source: defectdojo/templates/django-ingress.yaml
> apiVersion: networking.k8s.io/v1beta1
> kind: Ingress
> metadata:
>   name: release-name-defectdojo
>   labels:
>     defectdojo.org/component: django
>     app.kubernetes.io/name: defectdojo
>     app.kubernetes.io/instance: release-name
>     app.kubernetes.io/managed-by: Helm
>     helm.sh/chart: defectdojo-1.6.89
> spec:
>   tls:
>   - hosts:
>     - defectdojo.default.minikube.local
>     secretName: defectdojo-tls
>   rules:
>   - host: defectdojo.default.minikube.local
>     http:
>       paths:
>         - path: /
>           backend:
>             serviceName: release-name-defectdojo-django
>             servicePort: http
> ```
> 
> With my change, I get the correct Ingress resource:
> 
> ```
> # fixing_helm_ingress_capabilities_check branch
> $ helm template helm/defectdojo -s templates/django-ingress.yaml
> ---
> # Source: defectdojo/templates/django-ingress.yaml
> apiVersion: networking.k8s.io/v1
> kind: Ingress
> metadata:
>   name: release-name-defectdojo
>   labels:
>     defectdojo.org/component: django
>     app.kubernetes.io/name: defectdojo
>     app.kubernetes.io/instance: release-name
>     app.kubernetes.io/managed-by: Helm
>     helm.sh/chart: defectdojo-1.6.90-dev
> spec:
>   tls:
>   - hosts:
>     - defectdojo.default.minikube.local
>     secretName: defectdojo-tls
>   rules:
>   - host: defectdojo.default.minikube.local
>     http:
>       paths:
>         - pathType: Prefix
>           path: "/"
>           backend:
>             service:
>               name: release-name-defectdojo-django
>               port:
>                 name: http
> ```

Sorry for my late response, you can't validate `Capabilities` by using template, because it never reaches the k8s API, try with --dry-run  
Another concern I'm seeing, from version 1.7 it will be always true, because networking.k8s.io/v1 is there from version 1.7 when network policies have graduated

**Comment by mtesauro** [comment]
Closed as stale

**Pull request closed without merging**